### PR TITLE
Adding Rubocop

### PR DESCRIPTION
**Pull request by awead**
My :christmas_tree: :gift: to you... @cam156 @hectorcorrea.

Some of the files in the exclude list of rubocop.yml, especially those under Metrics/CyclomaticComplexity and PerceivedComplexity would be good candidates for refactoring. For now, I've put them on the exclusion list to get us to a passing state.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -85,10 +82,9 @@ def user_logged_in?
     end
 
     def has_access?
-      unless current_user && current_user.ldap_exist?
-        logger.error "User: `#{current_user.user_key}' does not exist in ldap"
-        render template: '/error/401', layout: "error", formats: [:html], status: 401
-      end
+      return if current_user && current_user.ldap_exist?
```
Comment (hectorcorrea) [inline app/controllers/application_controller.rb:85]: I love how the `if` statement is much clearer than the `unless` statement.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
 
-  before_action :get_linkedin_url, only: :show
+  before_action :linkedin_url, only: :show
 
-  def get_linkedin_url
-    @linkedInUrl = @user.linkedin_handle
-    @linkedInUrl = "http://www.linkedin.com/in/" + @linkedInUrl unless @linkedInUrl.blank? || @linkedInUrl.include?('linkedin.com')
-    @linkedInUrl = "http://" + @linkedInUrl unless @linkedInUrl.blank? || @linkedInUrl.include?('http')
+  def linkedin_url
+    @linkedin_url ||= format_linkedin_url
```
Comment (hectorcorrea) [inline app/controllers/users_controller.rb:7]: :+1:

File: app/models/migrate_audit.rb [hunk 3]
```diff
@@ -20,14 +20,12 @@ def self.f4_audit(auditor)
     end
   end
 
-  private
```
Comment (hectorcorrea) [inline app/models/migrate_audit.rb:23]: Why was the `private` removed?

File: config/application.rb [hunk 4]
```diff
@@ -34,13 +34,13 @@ def get_vhost_by_host
     def google_analytics_id
       vhost = get_vhost_by_host[0]
       ga_id = config.ga_host_map[vhost]
-      ga_id ||= Rails.application.secrets.google_analytics_tracking_id
+      ga_id || Rails.application.secrets.google_analytics_tracking_id
```
Comment (hectorcorrea) [inline config/application.rb:37]: Good catch!

File: spec/support/user_login.rb [hunk 5]
```diff
@@ -6,7 +6,6 @@ def login_as(login)
                                      respect_data_method: true,
                                      headers: { 'REMOTE_USER' => login })
     end
-    user = User.find_or_create_by(login: login)
```
Comment (hectorcorrea) [inline spec/support/user_login.rb:9]: Is it OK to remove this, isn't this creating the user if it is not there? 

Perhaps we don't need this during tests(?)

**Comment by hectorcorrea** [comment]
Awesome job @awead

File: spec/models/generic_file_spec.rb [hunk 6]
```diff
@@ -49,25 +49,24 @@
...
 
   describe "rescues ldp gone" do
-
-    let(:mint_service) { double}
+    let(:mint_service) { double }
 
     before do
       # create a file that is now gone
-      file = described_class.create { |file| file.apply_depositor_metadata('dmc') }
+      file = described_class.create { |f| f.apply_depositor_metadata('dmc') }
```
Comment (carolyncole) [inline spec/models/generic_file_spec.rb:56]: @awead I'm not sure I would agree with this change.  I wonder why rubocop felt the need...

**Comment by carolyncole** [comment]
@awead Thanks!  @hectorcorrea you ok with merging?

**Comment by hectorcorrea** [comment]
@cam156 abolutely, merge it away. I didn't merge it because I have not been working with SS recently and I didn't want to mess anything that might be in the works that I am now aware of.

**Comment by awead** [comment]
thanks! @hectorcorrea @cam156 I just do what the :cop: says... don't ask questions! :smile: But i'm open to tweaking the rules later.

**Pull request merged**